core/src/gauge.cc

```diff
@@ -1,35 +1,30 @@
 #include "prometheus/gauge.h"
 
+#include <atomic>
 #include <ctime>
 
 namespace prometheus {
 
-Gauge::Gauge(const double value) : value_{value} {}
+Gauge::Gauge(const double& value) : value_(value) {}
 
-void Gauge::Increment() { Increment(1.0); }
-
-void Gauge::Increment(const double value) {
-  if (value < 0.0) {
-    return;
-  }
-  Change(value);
+void Gauge::Increment(const double& value, const bool& alert) {
+  if (value < 0.0) return;
```

Comment:
While I think this line and the `Decrement` check makes sense, I also wonder if this check is necessary.  Like if the user wants chooses to decrement by a negative number, there won't be any warning or error raised, so would it be better to allow the user to do so?

Reply:
I agree but I'm not sure that warrants changing the original behaviour or not (the more that we leave the same, the easier it will be for merge updates).